src/evaluation/searcher.h

```diff
@@ -520,8 +532,13 @@ class Searcher {
     syzygy::WdlResult m_wdl { syzygy::WdlResultTableNotActive };
     uint8_t m_dtz {};
 
-    BitBoard m_board;
-    uint64_t m_hash;
+    struct StackInfo {
```

Comment:
Is it conventional to define a struct within a class?

I'm not used to seeing this, and we're already namespacing the other structs used by the searcher.

I wonder whether we should consistently move helper structs either inside or outside. What do you think?

Reply:
Depends on ownership. In this case the type belongs to the class why it's defined there.
Namespaces expose/group types - classes _can_ do the same. 

Notice that the struct is defined privately meaning that the type can only be used by the class itself :)

Reply:
I like the explicitness - thanks for clarifying :)